packages/vite/src/node/optimizer/index.ts

```diff
@@ -840,6 +836,30 @@ export async function addManuallyIncludedOptimizeDeps(
   }
 }
 
+function createOptimizeDepsIncludeResolver(
+  config: ResolvedConfig,
+  ssr: boolean,
+) {
+  const resolve = config.createResolver({
+    asSrc: false,
+    scan: true,
+    ssrOptimizeCheck: ssr,
+    ssrConfig: config.ssr,
+  })
+  return async (id: string) => {
+    const lastArrowIndex = id.lastIndexOf('>')
+    if (lastArrowIndex === -1) {
+      return await resolve(id, undefined, undefined, ssr)
+    }
+    // split nested selected id by last '>', for example:
+    // 'foo > bar > baz' => 'foo > bar' & 'baz'
+    const nestedRoot = id.substring(0, lastArrowIndex).trim()
+    const nestedPath = id.substring(lastArrowIndex + 1).trim()
+    const basedir = nestedResolveFrom(nestedRoot, config.root, false, ssr)
```

Comment:
```suggestion
    const basedir = nestedResolveFrom(
      nestedRoot,
      config.root,
      config.resolve.preserveSymlinks,
      ssr,
    )
```
I think we should pass `preserveSymlinks` because we were passing it before.

Reply:
Good point. I thought this was false because we didn't set `preserveSymlinks` in `createResolver`, but it looks like it falls back to `config.resolve.preserverSymlinks` if unset.